Wastewater_Seq_analysis/freyja_lin_agg_to_dict_long_df_20220603.py

```diff
@@ -0,0 +1,147 @@
+
+
+#print(new_agg_df_all_dict.index)
+
+#For direct freyja results
+#Shorten the sample names to remove '_out_variants.tsv'
+new_agg_df_all_dict.index = [x.split('_out')[-2] for x in new_agg_df_all_dict.index]
+#print(new_agg_df_all_dict.index)
+
+#Get the lineage dictionary and convert dictionary to df
+my_lin_dict = new_agg_df_all_dict.loc[:, 'linDict']
+new_df_dict = pd.DataFrame(my_lin_dict, index=None)
+
+
+new_df_dict.reset_index(inplace=True)
+
+#print(new_df_dict)
+
+#Creating matrix from lineage dictionary
+new_df_dict_matrix = pd.DataFrame(new_df_dict['linDict'].values.tolist(), index=new_df_dict['index'])
+
+#print(new_df_dict_matrix)
+
+round(new_df_dict_matrix, 3)
+
+new_df_dict_matrix.reset_index(inplace=True)
+
+
+#edit index names if needed, replace "-" with "_"
+new_df_dict_matrix.loc[:, 'index'] = new_df_dict_matrix['index'].apply(lambda x: x.replace("-", "_"))
+
+#melting matrix to long df
+new_df_dict_matrix_melt = pd.melt(new_df_dict_matrix,id_vars=['index'],var_name='lineages', value_name='abundance')
+
+new_df_dict_matrix_melt_noNA = new_df_dict_matrix_melt.dropna(axis=0, how='any')
+
+#Save the long df output
+new_df_dict_matrix_melt_noNA.to_csv(f'{run_name}_freyja_lin_agg_merged_dict_long_df.csv', sep=',')
+
+#Below are the optional steps to add an additional column of 'Lineages' for 'sub-lineages' in the data output
+uniq_lineage_list = new_df_dict_matrix_melt_noNA.lineages.unique()
+
+np.savetxt('freyja_uniq_lineage_list.csv', uniq_lineage_list, fmt='%s', delimiter = ",")
+
+
+uniq_lineage = pd.read_csv("freyja_uniq_lineage_list.csv",
+                 names=["lineages"])
+
+#Adding parent lineage column for individual runs
+uniq_lineage['Parent_lineage'] = np.where(uniq_lineage.lineages.str.startswith('BA.3'), 'Omicron BA.3',
+                                np.where(uniq_lineage.lineages.str.startswith('BA.2'), 'Omicron BA.2',
+                                np.where(uniq_lineage.lineages.str.startswith('BA.1'), 'Omicron BA.1',
+                                np.where(uniq_lineage.lineages.str.startswith('B.1.1.529'), 'Omicron',
+                                np.where(uniq_lineage.lineages.str.startswith('BA.4'), 'Omicron BA.4',
+                                np.where(uniq_lineage.lineages.str.startswith('BA.5'), 'Omicron BA.5',
+                                np.where(uniq_lineage.lineages.str.startswith('AP'), 'B.1.1',
+                                np.where(uniq_lineage.lineages.str.startswith('AV'), 'B.1.1',
+                                np.where(uniq_lineage.lineages.str.startswith('AK'), 'B.1.1',
+                                np.where(uniq_lineage.lineages.str.startswith('AM'), 'B.1.1',
+                                np.where(uniq_lineage.lineages.str.startswith('AZ'), 'B.1.1',
+                                np.where(uniq_lineage.lineages.str.startswith('AT'), 'B.1.1',
+                                np.where(uniq_lineage.lineages.str.startswith('P.'), 'B.1.1',
```

Comment:
I hesitate to group gamma with B.1.1 lineages

Reply:
I can modify that. My focus was to keep the recent lineages updated.